### PR TITLE
custom light sabers, bigger hitboxes and more small improvements

### PR DESCRIPTION
**Pull request by leandrodreamer**
- adding the ability to add custom sabers (only esthetics)

https://user-images.githubusercontent.com/26260718/114970895-edd22580-9e40-11eb-9c6e-cde90ca86ba2.mp4
(with that added a test saber with some particles effect)

- i tried making the blocks hit boxes bigger and i think they feel better now, unfair misses feeling is way less frequent

imagen

- also changed the percent of the right from a shader to a proper texture progress

### CODE REVIEW

**Comment by Muvkina** [comment]
@leandrodreamer Awesome work! I noticed the hitboxes were off too. For reference though here is the actual hitbox for beatsaber. [Beatsaber Hitbox](https://bsaber.com/wp-content/uploads/2019/02/image-2.png)

**Comment by leandrodreamer** [comment]
added the ability to set different cut sounds for the sabers

https://user-images.githubusercontent.com/26260718/114987529-e455b700-9e5a-11eb-9712-1c27c0e60c52.mp4

**Comment by Muvkina** [comment]
> added the ability to set different cut sounds for the sabers
> bssaber2_a.mp4

Awesome! It would be cool to add volume sliders for hit sounds and song volume.

**Comment by Firepal** [comment]
Oh this is gonna be fun

File: game/settings_Panel.gd [hunk 1]
```diff
@@ -94,13 +106,6 @@ func _on_right_sable_col_color_changed(color,overwrite=true):
 
 
 func _on_saber_tail_toggled(button_pressed,overwrite=true):
-	if button_pressed:
-		game.left_saber.set_tail_size(3)
-		game.right_saber.set_tail_size(3)
-	else:
-		game.left_saber.set_tail_size(0)
-		game.right_saber.set_tail_size(0)
-
```
Comment (Firepal) [inline game/settings_Panel.gd:103]: The "Saber Tail" isn't togglable anymore. Remember to reimplement this (probably using a bool in default_saber :)

**Comment by leandrodreamer** [comment]
imagen
imagen
made the cubes brighter so it's easier to identify the color from far away

**Comment by leandrodreamer** [comment]
accidentally closed the pr, sorry abt that

**Comment by leandrodreamer** [comment]
fixed the trail settings, added some new particles when cutting for the particle sword saber

also i make the cubes slightly brighter so they don't look black when they're far away


https://user-images.githubusercontent.com/26260718/115088680-56b7ad00-9ed6-11eb-8cd6-785306a7e4c5.mp4

**Comment by leandrodreamer** [comment]
i was testing an idea to sync the cut sound when you cut the cubes a little too early or late, should i include it here? (its pretty simple but makes gameplay feels better)

https://user-images.githubusercontent.com/26260718/118380606-b32aec80-b5a8-11eb-9c37-a52c7b176fc9.mp4

**Comment by NeoSpark314** [comment]
Both is fine (include here or make a separate PR); I think this will make a big impact on the "feel" of the game.

**Comment by leandrodreamer** [comment]
i add a second area to the cubes to have a separated collider, so when the saber match it uses the big collider, but when it doesn't it uses the small one, this will help when there are two cubes too close that u have to cut at  the same time to prevent both getting cutted by the same saber due to the big hitboxes
Screenshot-20210516153514-534x427

https://user-images.githubusercontent.com/26260718/118412030-635a2d00-b65d-11eb-86f0-3ab9863583f0.mp4

and with this, i think this PR its pretty much ready

**Pull request merged**